express-notes-api/server.js

```diff
@@ -0,0 +1,139 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+// gets a list of notes in an array
+app.get('/api/notes', async (req, res) => {
+  try {
+    const data = await fetchFile();
+    const array = [];
+
+    for (const note in data.notes) {
+      array.push(data.notes[note]);
+    }
+    res.status(200).json(array);
+  } catch (error) {
+    res.status(500).json({ error: 'Error reading file' });
+    console.error(error);
+  }
+});
+
+// gets a single note by id
+app.get('/api/notes/:id', async (req, res) => {
+  try {
+    const data = await fetchFile();
+    const id = req.params.id;
+
+    if (idErrorHandler(id, data, res)) return;
+    res.status(200).json(data.notes[id]);
+  } catch (error) {
+    res.status(500).json({ error: 'Error reading file' });
+    console.log(error);
+  }
+});
+
+// post a new note
+app.post('/api/notes', async (req, res) => {
+  try {
+    const data = await fetchFile();
+    const body = req.body;
+    const id = data.nextId;
+
+    if (Object.keys(body).length === 0) {
+      res.status(400).json({ error: 'content is a required field' });
+      return;
+    }
+
+    try {
```

Comment:
I just noticed this nested try/catch. Why are you nesting them? You need a try/catch around every `await`, but it doesn't have to be a separate try/catch for each one. You can have one single try/catch with multiple `await`s inside it, and that will work.

So the rule is that every `await` needs to be inside a try/catch _at some level_.